### DISCUSSION TITLE
Preview of a Scene by showing frames as thumbnails images.

### DISCUSSION THREAD

**Discussion by kolibril13** [ManimCommunity/manim#2779] [Suggestions and Proposals]
For building up complex scenes, It might be useful to not render the whole video but to only have a preview at some frames of it.
To give an example, this Video:

https://user-images.githubusercontent.com/44469195/108001842-ec75af80-6fed-11eb-95c7-4e19b3c2c211.mp4

Could also be displayed as this image:
![image](https://user-images.githubusercontent.com/44469195/108001811-d5cf5880-6fed-11eb-8eac-cb090200fdd4.png)

I can think of a lot of use cases where this could be helpful. Also with this method, we could have video-tests without building a video-test infrastructure.
This could be implemented as a new flag, e.g. `-thumbnails`

**Comment by WampyCakes** [comment]
Sounds like a great idea to me

**Comment by kolibril13** [comment]
Okay, here is a first idea to do so:
First, rendering a scene to pngs like this:
```
with "manim ExampleScene -qm -v WARNING --disable_caching  -g" in the console:
class ExampleScene(Scene):
    def construct(self):
        circle = Circle()
        square = Square()
        square.flip(RIGHT)
        square.rotate(-3 * TAU / 8)
        circle.set_fill(PINK, opacity=0.5)

        self.play(ShowCreation(square))
        self.play(Transform(square, circle))
```

and then adding the preview scene like this:

```py
class PreviewWindows(Group):
    def __init__(self):
        Group.__init__(self)
        height= 9/4
        width = 16/4
        frames = [Rectangle(height=height,width=width).set_color(BLUE) for i in range(0,6)]
        a=VGroup(*frames[0:3]).set_x(0).arrange(buff=0.5).shift(UP)
        b=VGroup(*frames[3:6]).set_x(0).arrange(buff=0.5).shift(2 * DOWN)
        self.add(a,b)
        imgs= Group()
        total_frames= 60
        previews= len(frames)
        for i, frame in enumerate(frames):
            position = int(total_frames/previews*i)
            img= ImageMobject(f"media/images/first_point/ExampleScene{position}.png").move_to(frame).scale(0.42)
            imgs.add(img)
        self.add(imgs)
        
class FameIdeaNo1(Scene):
    def construct(self):
        pw= PreviewWindows()
        self.add(pw)
```
gives this output:
![image](https://user-images.githubusercontent.com/44469195/109326948-10d85400-7858-11eb-9155-1f0581469deb.png)

Maybe we should wait with this feature until the click pr is merged:  #1013

**Comment by jsonvillanueva** [comment]
> Maybe we should wait with this feature until the click pr is merged: #1013

Thanks for the consideration, but the logic for the flag can be easily swapped into #1013 later so if someone wants to work on this before that's merged/closed, feel free.

**Comment by huguesdevimeux** [comment]
That's a nice idea, I like it.
